app/src/org/commcare/adapters/EntityStringFilterer.java

```diff
@@ -83,7 +83,9 @@ private void buildMatchList() {
             }
             Entity<TreeReference> e = fullEntityList.get(index);
             if (isCancelled()) {
-                break;
```

Comment:
Just realized this is a break, not a return, which makes me question a bit again whether I quite understand what's going on, since a break should _definitely_ still resulted in a succesful/end transaction pair being called, but this change should be an improvement either way.